video/out/w32_common.c

```diff
@@ -1222,33 +1286,41 @@ static void *gui_thread(void *ptr)
 
     thread_disable_ime();
 
-    WNDCLASSEXW wcex = {
-        .cbSize = sizeof wcex,
-        .style = CS_HREDRAW | CS_VREDRAW,
-        .lpfnWndProc = WndProc,
-        .hInstance = HINST_THISCOMPONENT,
-        .hIcon = LoadIconW(HINST_THISCOMPONENT, L"IDI_ICON1"),
-        .hCursor = LoadCursor(NULL, IDC_ARROW),
-        .lpszClassName = classname,
-    };
-    RegisterClassExW(&wcex);
-
     w32_thread_context = w32;
 
     if (w32->opts->WinID >= 0)
         w32->parent = (HWND)(intptr_t)(w32->opts->WinID);
 
+    ATOM cls = get_window_class();
     if (w32->parent) {
         RECT r;
         GetClientRect(w32->parent, &r);
-        w32->window = CreateWindowExW(WS_EX_NOPARENTNOTIFY, classname,
-                                      classname,
+        w32->window = CreateWindowExW(WS_EX_NOPARENTNOTIFY,
+                                      (LPWSTR)MAKEINTATOM(cls), L"mpv",
                                       WS_CHILD | WS_VISIBLE,
                                       0, 0, r.right, r.bottom,
                                       w32->parent, 0, HINST_THISCOMPONENT, NULL);
+
+        if (w32->window) {
+            // Install a hook to get notifications when the parent changes size
+            DWORD pid;
+            DWORD tid = GetWindowThreadProcessId(w32->parent, &pid);
+
+            // If the parent window lives inside the current process (libmpv,)
```

Comment:
Stray `,`?

Also might be better to move all this code into a separate function.